File: kredi-app-client/src/app/core/kredi-urunu.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { API_BASE_URL } from './api-config';

export interface KrediUrunu {
  id: number;
  ad: string;
  minVade: number;
  maksVade: number;
  faizOrani: number;
  kkdf: number;
  bsmv: number;
  minTutar: number;
  maksTutar: number;
}

export type KrediUrunuFormu = Omit<KrediUrunu, 'id'>;

@Injectable({
  providedIn: 'root',
})
export class KrediUrunuService {
  private readonly apiUrl = `${API_BASE_URL}/KrediUrunu`;

  constructor(private http: HttpClient) {}

  tumunuGetir(): Observable<KrediUrunu[]> {
    return this.http.get<KrediUrunu[]>(this.apiUrl);
  }

  getir(id: number): Observable<KrediUrunu> {
    return this.http.get<KrediUrunu>(`${this.apiUrl}/${id}`);
  }

  olustur(urun: KrediUrunuFormu): Observable<KrediUrunu> {
    return this.http.post<KrediUrunu>(this.apiUrl, urun);
  }

  guncelle(id: number, urun: KrediUrunuFormu): Observable<void> {
    return this.http.put<void>(`${this.apiUrl}/${id}`, { id, ...urun });
  }

  sil(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

File: KrediApp.Data/Entities/KrediUrunu.cs
using System;
using System.Collections.Generic;

namespace KrediApp.Data.Entities;

public partial class KrediUrunu
{
    public int Id { get; set; }

    public string Ad { get; set; } = null!;

    public int MinVade { get; set; }

    public int MaksVade { get; set; }

    public decimal FaizOrani { get; set; }

    public decimal Kkdf { get; set; }

    public decimal Bsmv { get; set; }

    public decimal MinTutar { get; set; }

    public decimal MaksTutar { get; set; }

    public virtual ICollection<KrediBasvuru> KrediBasvurus { get; set; } = new List<KrediBasvuru>();
}
